tests/expressions/typing/conftest.py

```diff
@@ -97,6 +97,12 @@
 ALL_DATATYPES_BINARY_PAIRS += ALL_TEMPORAL_DATATYPES_BINARY_PAIRS
 
 
+DECIMAL_DTYPES = [
+    (DataType.decimal128(7, 2), pa.array([1, 2, None], type=pa.decimal128(7, 2))),
+    (DataType.decimal128(4, 3), pa.array([1, 2, None], type=pa.decimal128(4, 3))),
```

Comment:
Also put some examples in here where the sum results in a number that requires a larger precision

Reply:
I also agree that we could have some correctness tests for decimal as well

Reply:
since here is mostly testing the type resolution, I've put the case for precision change in the new test_decimals test - which I think is the natural home for testing the math functionality